ciao-image/datastore/posix.go

```diff
@@ -44,5 +44,12 @@ func (p *Posix) Write(ID string, body io.Reader) (int64, error) {
 func (p *Posix) Delete(ID string) error {
 	imageName := path.Join(p.MountPoint, ID)
 
-	return os.Remove(imageName)
+	_, err := os.Stat(imageName)
+	if !os.IsNotExist(err) && err != nil {
```

Comment:
I'm not sure the logic is correct here.  If os.IsNotExist returns true, i.e., the file is known to not exist we then proceed to try and delete it which doesn't make any sense.  You probably want

```
    if err != nil {
        if os.IsNotExist(err) {
            err  = nil
       }
      return err
   }
```